corehq/apps/consumer_user/tasks.py

```diff
@@ -0,0 +1,40 @@
+from django.conf import settings
+from django.core.signing import TimestampSigner
+from django.template.loader import render_to_string
+from django.urls import reverse
+from django.utils.encoding import force_bytes
+from django.utils.http import urlsafe_base64_encode
+from django.utils.translation import ugettext_lazy as _
+
+from celery.task import task
+
+from dimagi.utils.web import get_url_base
+
+from corehq.apps.consumer_user.models import ConsumerUserInvitation
+from corehq.apps.hqcase.utils import update_case
+from corehq.apps.hqwebapp.tasks import send_html_email_async
+
+from .const import CONSUMER_INVITATION_SENT, CONSUMER_INVITATION_STATUS
+
+
+@task
+def create_new_invitation(case_id, domain, opened_by, email):
+    invitation = ConsumerUserInvitation.create_invitation(case_id, domain, opened_by, email)
+    url = '%s%s' % (get_url_base(),
```

Comment:
FYI I get this error when running this in celery:
```
2021-03-02 21:23:54,886: INFO/MainProcess] Received task: corehq.apps.consumer_user.tasks.create_new_invitation[1d1e586c-7ef7-498d-acd6-61372c937621]  
[2021-03-02 21:23:54,920: ERROR/ForkPoolWorker-6] Task corehq.apps.consumer_user.tasks.create_new_invitation[1d1e586c-7ef7-498d-acd6-61372c937621] raised unexpected: NoReverseMatch("'consumer_user' is not a registered namespace")
Traceback (most recent call last):
  File "/home/frener/.virtualenvs/cchq/lib/python3.8/site-packages/django/urls/base.py", line 75, in reverse
    extra, resolver = resolver.namespace_dict[ns]
KeyError: 'consumer_user'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/frener/.virtualenvs/cchq/lib/python3.8/site-packages/celery/app/trace.py", line 375, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/home/frener/.virtualenvs/cchq/lib/python3.8/site-packages/celery/app/trace.py", line 632, in __protected_call__
    return self.run(*args, **kwargs)
  File "/home/frener/dev/dimagi/commcare-hq/corehq/apps/consumer_user/tasks.py", line 24, in create_new_invitation
    reverse('consumer_user:consumer_user_register',
  File "/home/frener/.virtualenvs/cchq/lib/python3.8/site-packages/django/urls/base.py", line 86, in reverse
    raise NoReverseMatch("%s is not a registered namespace" % key)
django.urls.exceptions.NoReverseMatch: 'consumer_user' is not a registered namespace
```

Reply:
Its working in my local system when I trigger the call directly.
```
[2021-03-03 10:19:33,850: INFO/MainProcess] Received task: corehq.apps.consumer_user.tasks.create_new_invitation[1138ed38-8487-41f9-8df2-5ff10bdd453c]  
[2021-03-03 10:19:34,204: WARNING/ForkPoolWorker-6] http://localhost:8000/hq/consumer_user/signup/MjQ:1lHObN:-8hAS0-6c2CSKgckOVNlP60X5Z4/
[2021-03-03 10:19:34,248: INFO/ForkPoolWorker-6] Received Form a61e5f87f4a94f04ac0825ea3636fdae with 0 attachments
[2021-03-03 10:19:34,310: INFO/ForkPoolWorker-6] Finished normal processing for Form a61e5f87f4a94f04ac0825ea3636fdae
[2021-03-03 10:19:34,313: INFO/ForkPoolWorker-6] Task corehq.apps.consumer_user.tasks.create_new_invitation[1138ed38-8487-41f9-8df2-5ff10bdd453c] succeeded in 0.4627520860012737s: None
[2021-03-03 10:20:08,799: INFO/MainProcess] Received task: corehq.apps.consumer_user.tasks.create_new_invitation[e5fd1b81-a816-4b1f-96b0-13cd446b1678]  
[2021-03-03 10:20:09,169: WARNING/ForkPoolWorker-12] http://localhost:8000/hq/consumer_user/signup/MjU:1lHObw:voZOGBr9qQtJTYLfyIRFPf_DpUY/
[2021-03-03 10:20:09,203: INFO/ForkPoolWorker-12] Received Form eb1e4a8de3c54fbfb79780cc4bcaf3ae with 0 attachments
[2021-03-03 10:20:09,272: INFO/ForkPoolWorker-12] Finished normal processing for Form eb1e4a8de3c54fbfb79780cc4bcaf3ae
[2021-03-03 10:20:09,276: INFO/ForkPoolWorker-12] Task corehq.apps.consumer_user.tasks.create_new_invitation[e5fd1b81-a816-4b1f-96b0-13cd446b1678] succeeded in 0.4763715970002522s: None
[2021-03-03 10:20:15,488: INFO/MainProcess] Received task: corehq.apps.consumer_user.tasks.create_new_invitation[1edfe58b-76c2-431c-b6a8-22f52cbea709]  
[2021-03-03 10:20:15,863: WARNING/ForkPoolWorker-1] http://localhost:8000/hq/consumer_user/signup/MjY:1lHOc3:fodmjQtbPEgqF-8lyRJpa3NZ_04/
[2021-03-03 10:20:15,893: INFO/ForkPoolWorker-1] Received Form 5ebe229fde2e4ea68c22f4529541d82a with 0 attachments
[2021-03-03 10:20:15,951: INFO/ForkPoolWorker-1] Finished normal processing for Form 5ebe229fde2e4ea68c22f4529541d82a
[2021-03-03 10:20:15,955: INFO/ForkPoolWorker-1] Task corehq.apps.consumer_user.tasks.create_new_invitation[1edfe58b-76c2-431c-b6a8-22f52cbea709] succeeded in 0.466650219001167s: None
[2021-03-03 10:20:16,090: INFO/MainProcess] Received task: corehq.apps.consumer_user.tasks.create_new_invitation[33e3fcae-5e4a-4d3a-91b3-7a922718d73d]  
[2021-03-03 10:20:16,471: WARNING/ForkPoolWorker-2] http://localhost:8000/hq/consumer_user/signup/Mjc:1lHOc4:hVJ8QRhq3jWPmfuHlxua-1scLos/
[2021-03-03 10:20:16,504: INFO/ForkPoolWorker-2] Received Form cb44518343e446f39fcaa08b806da419 with 0 attachments
[2021-03-03 10:20:16,561: INFO/ForkPoolWorker-2] Finished normal processing for Form cb44518343e446f39fcaa08b806da419
[2021-03-03 10:20:16,565: INFO/ForkPoolWorker-2] Task corehq.apps.consumer_user.tasks.create_new_invitation[33e3fcae-5e4a-4d3a-91b3-7a922718d73d] succeeded in 0.4747796890005702s: None
```

Reply:
I checked if I missed anything while pushing. I pushed all the changes I made. Can you tell me how are you triggering this task? I will try to replicate in my system

Reply:
Weird, I was testing by creating a case through an app I built locally. I will try again and see if there is anything with my local setup that was causing this.